internal/venice-test-common/src/main/java/com/linkedin/venice/integration/utils/VeniceMultiClusterCreateOptions.java

```diff
@@ -0,0 +1,365 @@
+        .append("childControllerProperties:")
+        .append(childControllerProperties)
+        .append(", ")
+        .append("veniceProperties:")
+        .append(veniceProperties)
+        .append(", ")
+        .append("multiD2:")
+        .append(multiD2)
+        .append(", ")
+        .append("zk:")
+        .append(zkServerWrapper == null ? "null" : zkServerWrapper.getAddress())
+        .append(", ")
+        .append("kafka:")
+        .append(kafkaBrokerWrapper == null ? "null" : kafkaBrokerWrapper.getAddress())
+        .append(", ")
+        .append("kafkaClusterMap:")
+        .append(kafkaClusterMap)
+        .toString();
+  }
+
+  public VeniceMultiClusterCreateOptions(Builder builder) {
+    coloName = builder.coloName;
+    numberOfClusters = builder.numberOfClusters;
+    numberOfControllers = builder.numberOfControllers;
+    numberOfServers = builder.numberOfServers;
+    numberOfRouters = builder.numberOfRouters;
+    replicationFactor = builder.replicationFactor;
+    partitionSize = builder.partitionSize;
+    enableAllowlist = builder.enableAllowlist;
+    enableAutoJoinAllowlist = builder.enableAutoJoinAllowlist;
+    rebalanceDelayMs = builder.rebalanceDelayMs;
+    minActiveReplica = builder.minActiveReplica;
+    sslToStorageNodes = builder.sslToStorageNodes;
+    randomizeClusterName = builder.randomizeClusterName;
+    multiColoSetup = builder.multiColoSetup;
+    zkServerWrapper = builder.zkServerWrapper;
+    kafkaBrokerWrapper = builder.kafkaBrokerWrapper;
+    childControllerProperties = builder.childControllerProperties;
+    veniceProperties = builder.veniceProperties;
+    multiD2 = builder.multiD2;
+    forkServer = builder.forkServer;
+    kafkaClusterMap = builder.kafkaClusterMap;
+  }
+
+  public static class Builder {
+    private String coloName = "";
+    private final int numberOfClusters;
+    private int numberOfControllers = DEFAULT_NUMBER_OF_CONTROLLERS;
+    private int numberOfServers = DEFAULT_NUMBER_OF_SERVERS;
+    private int numberOfRouters = DEFAULT_NUMBER_OF_ROUTERS;
+    private int replicationFactor = DEFAULT_REPLICATION_FACTOR;
+    private int partitionSize = DEFAULT_PARTITION_SIZE_BYTES;
+    private int minActiveReplica;
+    private long rebalanceDelayMs = DEFAULT_DELAYED_TO_REBALANCE_MS;
+    private boolean enableAllowlist = false;
+    private boolean enableAutoJoinAllowlist = false;
+    private boolean sslToStorageNodes = DEFAULT_SSL_TO_STORAGE_NODES;
+    private boolean randomizeClusterName = true;
+    private boolean multiColoSetup = false;
+    private boolean multiD2 = false;
+    private boolean forkServer = false;
+    private boolean isMinActiveReplicaSet = false;
+    private Map<String, Map<String, String>> kafkaClusterMap;
+    private ZkServerWrapper zkServerWrapper;
+    private KafkaBrokerWrapper kafkaBrokerWrapper;
+    private Properties childControllerProperties;
+    private VeniceProperties veniceProperties;
+
+    public Builder(int numberOfClusters) {
```

Comment:
What's special about `numberOfClusters`? I think it be more readable if this parameter is specified via setter, like other parameters.

Reply:
This means that it is a mandatory argument.

Reply:
Got it. Thanks for explaining.